Thistle/galley-sdk
# VEG-261: make recipe details retrieval location aware

## Description
Please include a summary of the change. List any dependencies that are required for this pull request.
## Test Plan
1. In Galley staging, pull up a recipe like [Sesame Edamame Dip](https://staging-app.galleysolutions.com/recipes/cmVjaXBlOjE4MTY3Mg==?userLocationId=bG9jYXRpb246MTkyOA%3D%3D) that has base recipe, and a component with vendor items from multiple locations like [frozen edamame](https://staging-app.galleysolutions.com/ingredients/aW5ncmVkaWVudDoyNDQ0NDM=?userLocationId=bG9jYXRpb246MTkyOA%3D%3D)
2. Navigate to a location specific vendor item (West or East Coast), and under the nutrition tab, add unique dietary flags and nutrition data to the vendor item. For nutrition data it is helpful to pick a field that would be 0 otherwise, such as added sugar.
3. In the galley-sdk virtual environment, call `get_formatted_recipes_data` and `get_raw_recipes_data` with the recipe id and location name of the vendor item you modified. Example queries for Sesame Edamame Dip, Burlington:
```
import galley
from galley.queries import *
from galley.formatted_queries import *

get_formatted_recipes_data(recipe_ids=['cmVjaXBlOjE4MTY3Mg=='], location_name='Burlington')
get_raw_recipes_data(recipe_ids=['cmVjaXBlOjE4MTY3Mg=='], location_name='Burlington')
```
4. Verify that your modifications from Galley are present in the results. 
 - For the `get_formatted_recipes_data` result, the dietary flags you added should be listed as values for `allergens`, and the nutritional info in `nutrition`.  
 - For `get_raw_recipes_data`, the nutrition information you modified should be present in the `reconciledNutritionals` at the top most level of the data, as well as in the `reconciledNutritionals` info for the base recipe within the recipe. `dietaryFlagsWithUsages` should also be present at the top most level with your additional flag ids. 
5. Repeat these calls with the opposite location name, and verify that the return data matches what is in Galley for that location. 
## Versioning
0.43.0
